sam/exploration/signalaligner.py

```diff
@@ -0,0 +1,259 @@
+    signal_one : np.ndarray (default=None)
+    signal_two : np.ndarray (default=None)
+
+    Examples
+    --------
+    # Example 1
+    >>> import numpy as np
+    >>> from sam.exploration.signalaligner import SignalAligner
+
+    >>> signal_one = np.array([0, 1, 2, 3, 4, 3, 2, 1, 0])
+    >>> signal_two = np.array([0, 0, 0, 0, 0, 1, 2, 3, 4, 3, 2])
+    >>> offset, _ = SignalAligner.align_signals(signal_one, signal_two)
+
+    >>> print('Offset =', offset)
+
+    # Example 2
+    >>> N1 = 20
+    >>> N2 = 30
+    >>> N_aligned = 15
+
+    >>> lat = np.random.randn(N_aligned) + np.random.standard_normal(N_aligned)
+
+    >>> lat1 = np.random.randn(N1)
+    >>> i1 = np.random.randint(N1 - N_aligned)
+    >>> lat1[i1: i1 + N_aligned] = lat
+
+    >>> lat2 = np.random.randn(N2)
+    >>> i2 = np.random.randint(N2 - N_aligned)
+    >>> lat2[i2: i2 + N_aligned] = lat
+
+    >>> df1 = pd.DataFrame({'data': np.random.randn(N1), 'lat': lat1})
+    >>> df2 = pd.DataFrame({'data': np.random.randn(N2), 'lat': lat2})
+
+    >>> sa = SignalAligner()
+    >>> df, offset = sa.align_dataframes(df1, df2, col1, col2, reference=0)
+    """
+
+    def __init__(self, signal_one=None, signal_two=None):
+
+        self.signal_one = signal_one
+        self.signal_two = signal_two
+
+        if (signal_one is not None) and (signal_two is not None):
+            signal_one, signal_two = self._preprocess_signals(signal_one, signal_two)
+            self.offset, self.aligned_signal = self.align_signals(signal_one, signal_two)
+
+    def _preprocess_signals(self, signal_one, signal_two):
+        """Pad signals to have equal length.
+
+        NOTE: Assumes we have the same sampling frequency for both signals
+
+        Parameters
+        ----------
+        signal_one : np.ndarray
+        signal_two : np.ndarray
+
+        Returns
+        -------
+        signal_one_pp : np.ndarray
+        signal_two_pp : np.ndarray
+        """
+        N1 = len(signal_one)
+        N2 = len(signal_two)
+        if N1 < N2:
+            signal_one = self._zeropad(signal_one, N2)
+            self.signal_one = signal_one
+        elif N2 < N1:
+            signal_two = self._zeropad(signal_two, N1)
+            self.signal_two = signal_two
+
+        # we cannot allow nans for alignment.
+        signal_one = np.nan_to_num(signal_one, nan=0.0)
+        signal_two = np.nan_to_num(signal_two, nan=0.0)
+
+        return signal_one, signal_two
+
+    def _zeropad(self, signal, N):
+        """Zeropad signal to obtain N samples in total.
+        We pad the signal at the beginning.
+
+        Parameters
+        ----------
+        signal : np.ndarray
+            Input signals
+        N : int
+            Number of samples to pad to
+
+        Returns
+        -------
+        signal_pad : np.ndarray
+            Zero-padded input signal
+        """
+        signal_pad = np.concatenate([np.zeros((N - len(signal),)), signal])
+        return signal_pad
+
+    @staticmethod
```

Comment:
I added some preprocessing to the alignment function instead of the init now, so it's no longer a static method. I think what you meant was to take the function out of the class though